cachelib/allocator/CacheAllocator-inl.h

```diff
@@ -2763,12 +2633,118 @@ void CacheAllocator<CacheTrait>::evictForSlabRelease(
   }
 }
 
+template <typename CacheTrait>
+typename CacheAllocator<CacheTrait>::WriteHandle
+CacheAllocator<CacheTrait>::advanceIteratorAndTryEvictRegularItem(
+    MMContainer& mmContainer, EvictionIterator& itr) {
+  // we should flush this to nvmcache if it is not already present in nvmcache
+  // and the item is not expired.
+  Item& item = *itr;
+  const bool evictToNvmCache = shouldWriteToNvmCache(item);
+
```

Comment:
We still need this check that returns early if token is invalid. Otherwise we trigger the assert at L2677.

That said, we could just skip evicting to NvmCache on eviction if we couldn't acquire a valid PutToken (which can happen due to contention on the in-flight-puts queues). It may cause impact on hit rate. So it's best done as a separate change.

```
  // record the in-flight eviciton. If not, we move on to next item to avoid
  // stalling eviction.
  if (evictToNvmCache && !token.isValid()) {
    ++itr;
    stats_.evictFailConcurrentFill.inc();
    return WriteHandle{};
  }
```

Reply:
Got it, fixed.